File: util/name_util.py
# lapras/util/name_util.py
import re
from string import Template
from typing import Dict, Optional

from agent.agent_config import AgentConfig

class NameUtil:
    """
    Utility class for name expansion using templates.
    This is equivalent to the Java NameUtil class.
    """
    
    @staticmethod
    def expand(name: str, config: AgentConfig) -> str:
        """
        Expand variables in a template string using values from agent config.
        
        Args:
            name: The template string with placeholders in the format #{variable_name}
            config: The agent configuration containing variable values
            
        Returns:
            The expanded string with all variables replaced by their values
        """
        if not name:
            return ""
            
        # Find all variables in the template
        pattern = r'#\{([^}]+)\}'
        matches = re.finditer(pattern, name)
        
        # Create substitution map
        substitution_map = {}
        for match in matches:
            variable_name = match.group(1)
            value = config.get_option_or_default(variable_name, "")
            substitution_map[variable_name] = value
        
        # Replace variables with their values
        # We'll create a custom template class to handle our specific format
        class CustomTemplate(Template):
            delimiter = '#'
            pattern = r'\{([^}]*)\}'
            
        template = CustomTemplate(name)
        return template.safe_substitute(substitution_map)

File: lapras-middleware/src/main/java/kr/ac/kaist/cdsn/lapras/util/NameUtil.java
package kr.ac.kaist.cdsn.lapras.util;

import kr.ac.kaist.cdsn.lapras.AgentConfig;
import org.apache.commons.lang3.text.StrSubstitutor;

import java.util.HashMap;
import java.util.Map;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

/**
 * Created by Daekeun Lee on 2017-01-19.
 */
public class NameUtil {
    public static String expand(String name, AgentConfig config) {
        Map<String, String> substitutionMap = new HashMap<>();

        Pattern pattern = Pattern.compile("\\#\\{([^\\}]+)\\}");
        Matcher matcher = pattern.matcher(name);
        while(matcher.find()) {
            String variableName = matcher.group(1);
            String value = config.getOptionOrDefault(variableName, "");
            substitutionMap.put(variableName, value);
        }

        StrSubstitutor strSubstitutor = new StrSubstitutor(substitutionMap, "#{", "}");
        return strSubstitutor.replace(name);
    }
}
